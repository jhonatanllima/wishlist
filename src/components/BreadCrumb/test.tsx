import { render, screen } from '@testing-library/react';
import { renderWithTheme } from '~/utils/tests/helpers';

import { BreadCrumb } from '.';

const breadCrumbItems = [
  {
    title: 'Home',
    link: '/',
  },
];

const multiplesBreadCrumbItems = [
  {
    title: 'Home',
    separator: '>',
    link: '/',
  },

  {
    title: 'Favoritos',
    link: '/favoritos',
  },
];

describe('< BreadCrumb /> ', () => {
  it('should render correctly.', () => {
    const { container } = renderWithTheme(
      <BreadCrumb breadCrumbItems={breadCrumbItems} />
    );

    expect(screen.getByText(/Home/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly multiple breadcrumbs.', () => {
    const { container } = render(
      <BreadCrumb breadCrumbItems={multiplesBreadCrumbItems} />
    );

    expect(screen.getByText(/Home/i)).toBeInTheDocument();
    expect(screen.getByText(/Favoritos/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
