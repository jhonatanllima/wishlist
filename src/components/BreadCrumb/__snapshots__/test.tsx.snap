// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< BreadCrumb />  should render correctly multiple breadcrumbs. 1`] = `
.c0 {
  gap: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 a {
  gap: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000000;
  text-transform: capitalize;
}

.c0 strong {
  font-size: 2rem;
  text-transform: capitalize;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/"
    >
      <strong>
        Home
      </strong>
      <strong>
        &gt;
      </strong>
    </a>
    <a
      href="/favoritos"
    >
      <strong>
        Favoritos
      </strong>
      <strong />
    </a>
  </div>
</div>
`;

exports[`< BreadCrumb />  should render correctly. 1`] = `
.c0 {
  gap: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 a {
  gap: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #000000;
  text-transform: capitalize;
}

.c0 strong {
  font-size: 2rem;
  text-transform: capitalize;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  class="c0"
>
  <strong>
    Home
  </strong>
</div>
`;
